vms/rpcchainvm/runtime/subprocess/initializer.go

```diff
@@ -25,24 +27,46 @@ type initializer struct {
 	initialized chan struct{}
 }
 
-func newInitializer() *initializer {
+func newInitializer(path string) *initializer {
 	return &initializer{
+		path:        path,
 		initialized: make(chan struct{}),
 	}
 }
 
 func (i *initializer) Initialize(_ context.Context, protocolVersion uint, vmAddr string) error {
 	i.once.Do(func() {
 		if version.RPCChainVMProtocol != protocolVersion {
-			i.err = fmt.Errorf("%w. AvalancheGo version %s implements RPCChainVM protocol version %d. The VM implements RPCChainVM protocol version %d. Please make sure that there is an exact match of the protocol versions. This can be achieved by updating your VM or running an older/newer version of AvalancheGo. Please be advised that some virtual machines may not yet support the latest RPCChainVM protocol version",
```

Comment:
I do agree with your point that single-instance error shouldn't become a custom error object. The one exception that I can see here is that this error object carries multiple data elements.
Keeping these data elements decoupled from the string would allow us to log and index these ( which, in turn, would enable us to better our data-driven error mitigation )